test/TestBulkRenewal.js

```diff
@@ -0,0 +1,108 @@
+const ENS = artifacts.require('@ensdomains/ens/ENSRegistry');
+const HashRegistrar = artifacts.require('@ensdomains/ens/HashRegistrar');
+const PublicResolver = artifacts.require('@ensdomains/resolver/PublicResolver');
+const BaseRegistrar = artifacts.require('./BaseRegistrarImplementation');
+const ETHRegistrarController = artifacts.require('./ETHRegistrarController');
+const DummyOracle = artifacts.require('./DummyOracle');
+const StablePriceOracle = artifacts.require('./StablePriceOracle');
+const BulkRenewal = artifacts.require('./BulkRenewal');
+var Promise = require('bluebird');
+
+const namehash = require('eth-ens-namehash');
+const sha3 = require('web3-utils').sha3;
+const toBN = require('web3-utils').toBN;
+
+const NULL_ADDRESS = "0x0000000000000000000000000000000000000000"
+const ETH_LABEL = sha3('eth');
+const ETH_NAMEHASH = namehash.hash('eth');
+
+const advanceTime = Promise.promisify(function(delay, done) {
+    web3.currentProvider.send({
+        jsonrpc: "2.0",
+        "method": "evm_increaseTime",
+        params: [delay]}, done)
+    }
+);
+
+async function expectFailure(call) {
+    let tx;
+    try {
+        tx = await call;
+    } catch (error) {
+        // Assert ganache revert exception
+        assert.equal(
+            error.message,
+            'Returned error: VM Exception while processing transaction: revert'
+        );
+    }
+    if(tx !== undefined) {
+        assert.equal(parseInt(tx.receipt.status), 0);
+    }
+}
+
+contract('BulkRenewal', function (accounts) {
+    let ens;
+    let resolver;
+    let baseRegistrar;
+    let controller;
+    let priceOracle;
+    let bulkRenewal;
+
+    const secret = "0x0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF";
+    const ownerAccount = accounts[0]; // Account that owns the registrar
+    const registrantAccount = accounts[1]; // Account that owns test names
+
+    before(async () => {
```

Comment:
are we sure this shouldn't be a beforeEach? If we add more tests it may lead to later confusion because they will have a shared state.

Reply:
As above - this PR isn't about this file.